drivers/sensor/lsm6dsl/lsm6dsl.h

```diff
@@ -0,0 +1,712 @@
+
+#include <zephyr/types.h>
+#include <i2c.h>
+#include <misc/util.h>
+
+
+#define LSM6DSL_REG_FUNC_CFG_ACCESS			0x01
+#define LSM6DSL_MASK_FUNC_CFG_EN			BIT(7)
+#define LSM6DSL_SHIFT_FUNC_CFG_EN			7
+#define LSM6DSL_MASK_FUNC_CFG_EN_B			BIT(5)
+#define LSM6DSL_SHIFT_FUNC_CFG_EN_B			5
+
+#define LSM6DSL_REG_SENSOR_SYNC_TIME_FRAME		0x04
+#define LSM6DSL_MASK_SENSOR_SYNC_TIME_FRAME_TPH		(BIT(3) | BIT(2) | \
+							 BIT(1) | BIT(0))
+#define LSM6DSL_SHIFT_SENSOR_SYNC_TIME_FRAME_TPH	0
+
+#define LSM6DSL_REG_SENSOR_SYNC_RES_RATIO		0x05
+#define LSM6DSL_MASK_SENSOR_SYNC_RES_RATIO		(BIT(1) | BIT(0))
+#define LSM6DSL_SHIFT_SENSOR_SYNC_RES_RATIO		0
+
+#define LSM6DSL_REG_FIFO_CTRL1				0x06
+#define LSM6DSL_MASK_FIFO_CTRL1_FTH			(BIT(7) | BIT(6) | \
+							 BIT(5) | BIT(4) | \
+							 BIT(3) | BIT(2) | \
+							 BIT(1) | BIT(0))
+#define LSM6DSL_SHIFT_FIFO_CTRL1_FTH			0
+
+#define LSM6DSL_REG_FIFO_CTRL2				0x07
+#define LSM6DSL_MASK_FIFO_CTRL2_TIMER_PEDO_FIFO_EN	BIT(7)
+#define LSM6DSL_SHIFT_FIFO_CTRL2_TIMER_PEDO_FIFO_EN	7
+#define LSM6DSL_MASK_FIFO_CTRL2_TIMER_PEDO_FIFO_DRDY	BIT(6)
+#define LSM6DSL_SHIFT_FIFO_CTRL2_TIMER_PEDO_FIFO_DRDY	6
+#define LSM6DSL_MASK_FIFO_CTRL2_FIFO_TEMP_EN		BIT(3)
+#define LSM6DSL_SHIFT_FIFO_CTRL2_FIFO_TEMP_EN		3
+#define LSM6DSL_MASK_FIFO_CTRL2_FTH			(BIT(2) | BIT(1) | \
+							 BIT(0))
+#define LSM6DSL_SHIFT_FIFO_CTRL2_FTH			0
+
+#define LSM6DSL_REG_FIFO_CTRL3				0x08
+#define LSM6DSL_MASK_FIFO_CTRL3_DEC_FIFO_GYRO		(BIT(5) | BIT(4) | \
+							 BIT(3))
+#define LSM6DSL_SHIFT_FIFO_CTRL3_DEC_FIFO_GYRO		3
+#define LSM6DSL_MASK_FIFO_CTRL3_DEC_FIFO_XL		(BIT(2) | BIT(1) | \
+							 BIT(0))
+#define LSM6DSL_SHIFT_FIFO_CTRL3_DEC_FIFO_XL		0
+
+#define LSM6DSL_REG_FIFO_CTRL4				0x09
+#define LSM6DSL_MASK_FIFO_CTRL4_STOP_ON_FTH		BIT(7)
+#define LSM6DSL_SHIFT_FIFO_CTRL4_STOP_ON_FTH		7
+#define LSM6DSL_MASK_FIFO_CTRL4_ONLY_HIGH_DATA		BIT(6)
+#define LSM6DSL_SHIFT_FIFO_CTRL4_ONLY_HIGH_DATA		6
+#define LSM6DSL_MASK_FIFO_CTRL4_DEC_DS4_FIFO		(BIT(5) | BIT(4) | \
+							 BIT(3))
+#define LSM6DSL_SHIFT_FIFO_CTRL4_DEC_DS4_FIFO		3
+#define LSM6DSL_MASK_FIFO_CTRL4_DEC_DS3_FIFO		(BIT(2) | BIT(1) | \
+							 BIT(0))
+#define LSM6DSL_SHIFT_FIFO_CTRL4_DEC_DS3_FIFO		0
+
+#define LSM6DSL_REG_FIFO_CTRL5				0x0A
+#define LSM6DSL_MASK_FIFO_CTRL5_ODR_FIFO		(BIT(5) | BIT(4) | \
+							 BIT(3))
+#define LSM6DSL_SHIFT_FIFO_CTRL5_ODR_FIFO		3
+#define LSM6DSL_MASK_FIFO_CTRL5_FIFO_MODE		(BIT(2) | BIT(1) | \
+							 BIT(0))
+#define LSM6DSL_SHIFT_FIFO_CTRL5_FIFO_MODE		0
+
+#define LSM6DSL_REG_DRDY_PULSE_CFG_G			0x0B
+#define LSM6DSL_MASK_DRDY_PULSE_CFG_G_DRDY_PULSED	BIT(7)
+#define LSM6DSL_SHIFT_DRDY_PULSE_CFG_G_DRDY_PULSED	7
+#define LSM6DSL_MASK_DRDY_PULSE_CFG_G_INT2_WRIST_TILT	BIT(0)
+#define LSM6DSL_SHIFT_DRDY_PULSE_CFG_G_INT2_WRIST_TILT	0
+
+#define LSM6DSL_REG_INT1_CTRL				0x0D
+#define LSM6DSL_MASK_INT1_CTRL_STEP_DETECTOR		BIT(7)
+#define LSM6DSL_SHIFT_INT1_CTRL_STEP_DETECTOR		7
+#define LSM6DSL_MASK_INT1_CTRL_SIGN_MOT			BIT(6)
```

Comment:
I think this is the same as previous comment: the additional '+' get me disaligned.
It's ok in my editor